### ISSUE TITLE
Increase/decrease font size

### ISSUE THREAD

**Issue by scholtzan** [xi-editor/xi-mac#312]
In most text editors <kbd>cmd</kbd><kbd>+</kbd> and <kbd>cmd</kbd><kbd>-</kbd> offer a quick way to increase and decrease the font size. It would be cool if xi mac would also support this.

**Comment by ryanbloom** [comment]
For what it's worth, I made [a hacky implementation of this](https://github.com/ryanbloom/xi-mac/commit/96888c431b49f3902b507046edd75c95171170f1) a few weeks ago. It could take more work to do properly, though, since [the docs say not to use both file-based and RPC-based config](https://github.com/xi-editor/xi-editor/blob/master/docs/docs/config.md#rpc-based-config). Then again, maybe it's okay for this use case?

If we want these changes to persist across sessions, the core will first need a way to modify the config file. That functionality might be useful for other settings, like theme and indentation.

**Comment by cmyr** [comment]
Hm. I could imagine this working as a per-buffer override setting? It wouldn't persist between sessions right now, but it could when we get state restoration working.

**Comment by ryanbloom** [comment]
@cmyr You're right, that could work too. VSCode seems to do a per-window override, while Sublime Text and Atom modify the original settings.

I guess there are 2 different use cases:
* Quickly change your global default font size by a point or two, and keep it that way
* Make a file big temporarily to show it to someone else

Personally, I would find the global version a little more intuitive, and use the commands to reset it if I ever want to. But for people who do lots of presentations or screencasts, I could see an override being more convenient.

@scholtzan did you have either particular version in mind?

**Comment by cmyr** [comment]
There are two sort of annoying limitations of both these projects:

1. because we handle global config through the config file, providing other mechanisms for managing settings is kind of a pita; we'd want some way to do structured parsing and editing of the config file, or something, to keep these things in sync; we don't have this right now. At some point we could move to using a more gui-ful settings pane, but until then I'm not sure what makes the most sense.

2. For the _per buffer_ settings; there's another problem, with width measurement: afaik we don't track the current font/font-size when doing width measurement, which means we aren't really prepared to handle more than a single font/size combo at a time. This needs to get addressed at some point; if someone wanted a place to start it might be a reasonable one?
